www/src/components/Toc.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+const Toc = ({ data, className }) => {
+  const generateTree = (headings) => (headings?.items?.length
+    ? (
+      <ul className="pgn-doc__toc-list">
+        {headings.items.map(heading => (
+          <li key={heading.url}>
+            <a href={heading.url}>{heading.title}</a>
+            {!!heading.items && (
+            <ul className="pgn-doc__toc-list">
+              {generateTree(heading)}
+            </ul>
+            )}
+          </li>
+        ))}
+      </ul>
+    ) : 'No headings on the page');
```

Comment:
I think it may make sense to render nothing (i.e., `null`) in the sidebar if there are no headings on the page?

Reply:
I think yeah, it would be cleaner. Changed to render `null` (no `Contents` also) if no headings there